### PR TITLE
example_config: add initial documentation

### PR DESCRIPTION
**Pull request by jzelinskie**
I left the middleware undocumented because I'm not sure what we want to do there.

### CODE REVIEW

File: example_config.yaml [hunk 1]
```diff
@@ -1,49 +1,89 @@
...
+
+  # The network interface that will bind to an HTTP endpoint that can be
+  # scraped by an instance of the Prometheus time series database.
+  # For more info see: https://prometheus.io
+  prometheus_addr: 0.0.0.0:6880
+
+  # The maximum number of peers returned in an announce.
   max_numwant: 50
+
+  # The default number of peers returned in an announce.
```
Comment (mrd0ll4r) [inline example_config.yaml]: ...if the client provides an invalid `num_want` parameter, i.e. `<=0`

File: example_config.yaml [hunk 2]
```diff
@@ -1,49 +1,89 @@
...
   default_numwant: 25
 
+  # This block defines configuration for the tracker's HTTP interface.
+  # If you do not wish to run this, delete this section.
   http:
+    # The network interface that will bind to an HTTP server for serving
+    # BitTorrent traffic.
```
Comment (mrd0ll4r) [inline example_config.yaml]: We should probably make a comment about how to specify IPv6 addresses: In quotation marks, using square brackets.

File: example_config.yaml [hunk 3]
```diff
@@ -1,49 +1,89 @@
...
     addr: 0.0.0.0:6881
+
+    # When enabled, the IP address used to connect to the tracker overrides
+    # the value clients advertise.
```
Comment (mrd0ll4r) [inline example_config.yaml]: It's the other way around, isn't it?

File: example_config.yaml [hunk 4]
```diff
@@ -1,49 +1,89 @@
...
     allow_ip_spoofing: false
+
+    # The HTTP Header containing the IP address of the client.
+    # This is only necessary if using a reverse proxy.
```
Comment (mrd0ll4r) [inline example_config.yaml]: Also: Does this only work with `allow_ip_spoofing=true`?

Comment (jzelinskie) [inline example_config.yaml]: IIRC, if the client doesn't provide an IP we fallback to the RemoteAddr as a last resort.

File: example_config.yaml (hunk 3 above)
Comment (jzelinskie) [inline example_config.yaml]: yep, I wrote this wrong.

File: example_config.yaml (hunk 2 above)
Comment (jzelinskie) [inline example_config.yaml]: Yeah, that's an unfortunate side effect of YAML

File: example_config.yaml (hunk 1 above)
Comment (jzelinskie) [inline example_config.yaml]: Unrelated: is numwant = 0 actually a bad value?

Comment (mrd0ll4r) [inline example_config.yaml]: Oh, numwant is optional, hmm. You mean something like seeders actually announcing with numwant=0? Problem is: How do we distinguish that from `numwant` missing (=default)? Maybe we should make `numwant` a signed integer again, and set that to -1?

Comment (jzelinskie) [inline example_config.yaml]: As discussed on IRC, we could make the sanitization a function of the bittorrent package along with a NumWant type that has semantic meaning for the -1 value.

**Pull request merged**